tasks/process_flakes.py

```diff
@@ -101,78 +154,73 @@ def get_test_instances(
     )
     test_instances = list(
         TestInstance.objects.filter(
-            repo_commit_branch_filter
-            & (test_failed_filter | test_passed_but_flaky_filter)
+            upload_filter & (test_failed_filter | test_passed_but_flaky_filter)
         )
         .select_related("test")
         .all()
     )
     return test_instances
 
 
-def generate_flake_dict(repo_id: int) -> FlakeDict:
+def fetch_curr_flakes(repo_id: int) -> dict[str, Flake]:
     flakes = Flake.objects.filter(repository_id=repo_id, end_date__isnull=True).all()
-    flake_dict = dict()
-    for flake in flakes:
-        flake_dict[(flake.test_id, flake.reduced_error_id)] = flake
-    return flake_dict
+    return {flake.test_id: flake for flake in flakes}
 
 
-def update_passed_flakes(test_instance: TestInstance, flake: Flake) -> None:
-    flake.count += 1
-    flake.recent_passes_count += 1
+def create_flake(
+    test_instance: TestInstance,
+    repo_id: int,
+) -> tuple[Flake, DailyTestRollup | None]:
+    # retroactively mark newly caught flake as flaky failure in its rollup
+    rollup = DailyTestRollup.objects.filter(
+        repoid=repo_id,
+        date=test_instance.created_at.date(),
+        branch=test_instance.branch,
+        test_id=test_instance.test_id,
+    ).first()
+
+    if rollup:
+        rollup.flaky_fail_count += 1
+    else:
+        log.warning(
+            "Could not find rollup when trying to update its flaky fail count",
+            extra=dict(
+                repoid=repo_id,
+                testid=test_instance.test_id,
+                branch=test_instance.branch,
+                date=test_instance.created_at.date(),
+            ),
+        )
```

Comment:
Do we have to worry about creating a new roll-up if it doesn't exist here?

Reply:
no, because we assume it exists, i guess this should really be a `sentry_sdk.capture_exception` because i wanted to know if this is even a possibility, i don't think it has come up in the logs before, so it's probably safe to remove soon